[keycloak] Add autoscaling support and KUBE_PING documentation (#330)
codecentric/helm-charts: charts/keycloak/README.md

    value: "2"
```

#### KUBE_PING Service Discovery

> Recent versions of Keycloak include a new Kuberntes Native [KUBE_PING](https://github.com/jgroups-extras/jgroups-kubernetes) service discovery protocol. This requires a little more configuration than DNS_PING but it is not difficult through the Helm Chart.

Reviewer: ```suggestion
Recent versions of Keycloak include a new Kubernetes native [KUBE_PING](https://github.com/jgroups-extras/jgroups-kubernetes) service discovery protocol.
This requires a little more configuration than DNS_PING but can easily be achieved with the Helm chart.
```

As with DNS_PING some environment variables must be configured as follows:

```yaml
extraEnv: |
  - name: JGROUPS_DISCOVERY_PROTOCOL
    value: kubernetes.KUBE_PING
  - name: KUBERNETES_NAMESPACE
    valueFrom:
      fieldRef:
        apiVersion: v1
        fieldPath: metadata.namespace
  - name: CACHE_OWNERS_COUNT
    value: "2"
  - name: CACHE_OWNERS_AUTH_SESSIONS_COUNT
    value: "2"
```

> However the Keycloak Pods must also be able to `get` and `list` Pods in the namespace which can be configured as follows:

Reviewer: ```suggestion
However, the Keycloak Pods must also get RBAC permissions to `get` and `list` Pods in the namespace which can be configured as follows:
```

```yaml
rbac:
  create: true

>   rules:
>   RBAC rules for KUBE_PING
>     - apiGroups:

Reviewer: This line must be removed or commented out
```suggestion
  rules:
    - apiGroups:
```

        - ""
      resources:
        - pods
      verbs:
        - get
        - list
```

#### Autoscaling

> Due to the caches in Keycloak only replicating to a few nodes (2 in the example configuration above) and the limited controls around autoscaling built into Kubernetes it has historically been problematic to autoscale Keycloak. However in Kubernetes 1.18 [additional controls were introduced](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-configurable-scaling-behavior) which make it possible to scale down in a more controlled manner.

Reviewer: ```suggestion
Due to the caches in Keycloak only replicating to a few nodes (two in the example configuration above) and the limited controls around autoscaling built into Kubernetes, it has historically been problematic to autoscale Keycloak.
However, in Kubernetes 1.18 [additional controls were introduced](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-configurable-scaling-behavior) which make it possible to scale down in a more controlled manner.
```

> The example autoscaling configuration in the values file scales from 3 up to a maximum of 10 Pods using CPU utilization as the metric. Scaling up is done as quickly as required but scaling down is done at a maximum rate of one Pod per 5 minutes.

Reviewer: ```suggestion
The example autoscaling configuration in the values file scales from three up to a maximum of ten Pods using CPU utilization as the metric. Scaling up is done as quickly as required but scaling down is done at a maximum rate of one Pod per five minutes.
```
